### PR TITLE
Made Services and Tests

### PR DESCRIPTION
**Pull request by aces47**

### CODE REVIEW

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs [hunk 1]
```diff
@@ -14,21 +14,28 @@ public UserService(IIdentityUnitOfWork identityUnitOfWork, IMapper mapper) : bas
         {
         }
 
-        public Task<SignInResult> SignInAsync(SignInModel model)
+        public async Task<SignInResult> SignInAsync(SignInModel model)
         {
-            return UnitOfWork.SignInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, false);
+            var result = await _unitOfWork.SignInManager.PasswordSignInAsync(model.Email,
```
Comment (kuzmiich) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs:19]: Bad naming

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs [hunk 2]
```diff
@@ -14,21 +14,28 @@ public UserService(IIdentityUnitOfWork identityUnitOfWork, IMapper mapper) : bas
...
+            return result;
         }
 
-        public Task<IdentityResult> SignUpAsync(SignUpModel model)
+        public async Task<IdentityResult> SignUpAsync(SignUpModel model)
         {
-            var mappedEntity = Mapper.Map<User>(model);
+            var mappedEntity = _mapper.Map<User>(model);
+
+            var result = await _unitOfWork.UserManager.CreateAsync(mappedEntity, model.Password);
```
Comment (kuzmiich) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs:31]: bad naming

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs (hunk 1 above)
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs:19]: Where?

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs (hunk 2 above)
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Services/UserService.cs:31]: Where?

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IReadOnlyService.cs [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 using DIMS_Core.BusinessLayer.Models;
 
 namespace DIMS_Core.BusinessLayer.Interfaces
 {
-    public interface IReadOnlyService : IDisposable
+    public interface IReadOnlyService<TModel> : IDisposable
+        where TModel : class
     {
-        Task<VUserProfileModel[]> GetAll();
+        Task<IEnumerable<TModel>> GetAll();
```
[the remark this answers is not in the document]
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IReadOnlyService.cs:11]: Fixed

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs [hunk 4]
```diff
@@ -33,10 +33,14 @@ public class UserProfileModel
...
+        {
+            return left.FirstName == right.FirstName && left.LastName == right.LastName;
+        }
 
-        // TODO: Task # 15
-        // You have to implement operator != by comparing First and Last names
+        public static bool operator !=(UserProfileModel left, UserProfileModel right)
+        {
+            return left.FirstName != right.FirstName && left.LastName != right.LastName;
+        }
```
[the remark this answers is not in the document]
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs:44]: Deleted

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/IdentityService.cs [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
 
 namespace DIMS_Core.BusinessLayer.Services
 {
-    public abstract class IdentityService : IService
+    public abstract class IdentityService : IIdentityService
     {
-        protected readonly IMapper Mapper;
-        protected readonly IIdentityUnitOfWork UnitOfWork;
+        protected readonly IMapper _mapper;
+        protected readonly IIdentityUnitOfWork _unitOfWork;
```
[the remark this answers is not in the document]
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Services/IdentityService.cs:12]: Fixed

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs [hunk 6]
```diff
@@ -33,10 +33,5 @@ public class UserProfileModel
 
         public string MobilePhone { get; set; }
 
-        // TODO: Task # 14
-        // You have to implement operator == by comparing First and Last names
-
-        // TODO: Task # 15
-        // You have to implement operator != by comparing First and Last names
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs:40]: Return this TODO back.

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs (hunk 6 above)
Comment (aces47) [reply DIMS-Core/DIMS-Core.BusinessLayer/Models/UserProfileModel.cs]: return TODO to UserProfileModel

**Review by kuzmiich — APPROVED** [review]

**Pull request merged**